dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/handler/ExternalSourceEntryUriListHandler.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * This class will handle ExternalSourceEntryUriList and it'll create Item objects based on them
+ */
+@Component
+public class ExternalSourceEntryUriListHandler implements UriListHandler<Item> {
+
+    private List<RequestMethod> allowedRequestMethods = new LinkedList<>(Arrays.asList(RequestMethod.POST));
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @Autowired
+    private ExternalDataService externalDataService;
+
+    @Autowired
+    private CollectionService collectionService;
+
+    private static final Logger log = org.apache.logging.log4j.LogManager
+        .getLogger(ExternalSourceEntryUriListHandler.class);
+
+
+    @Override
+    public boolean supports(List<String> uriList, String method) {
+        if (!allowedRequestMethods.contains(RequestMethod.valueOf(method))) {
+            return false;
+        }
+        for (String string : uriList) {
+            if (!(StringUtils.contains(string, "api/integration/externalsources") &&
+                StringUtils.contains(string, "entryValues"))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean validate(Context context, HttpServletRequest request, List<String> uriList,
+                            Class clazz) throws AuthorizeException {
+        if (uriList.size() > 1) {
+            return false;
+        }
+        if (clazz != Item.class) {
+            return false;
+        }
+        String owningCollectionString = request.getParameter("owningCollection");
+        if (StringUtils.isBlank(owningCollectionString)) {
+            return false;
+        }
+        try {
+            Collection collection = collectionService.find(context, UUID.fromString(owningCollectionString));
+            if (collection == null) {
+                return false;
+            }
+            if (!authorizeService.isAdmin(context)) {
+                throw new AuthorizeException("Only admins are allowed to create items using external data");
+            }
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public Item handle(Context context, HttpServletRequest request, List<String> uriList)
+        throws SQLException, AuthorizeException {
+        ExternalDataObject dataObject = getExternalDataObjectFromUriList(uriList);
+
+        String owningCollectionUuid = request.getParameter("owningCollection");
+        Collection collection = null;
+        Item item = null;
+        try {
+            collection = collectionService.find(context, UUID.fromString(owningCollectionUuid));
+            item = externalDataService.createItemFromExternalDataObject(context, dataObject, collection);
+        } catch (AuthorizeException | SQLException e) {
+            log.error(e.getMessage(), e);
```

Comment:
Again, needs a more descriptive error message please.  In this instance we likely need to log something like "Could not create Item in Collection with UUID=[uuid]"